src/services/refactors/moveToFile.ts

```diff
@@ -926,15 +926,15 @@ function getRangeToMove(context: RefactorContext): RangeToMove | undefined {
        function add(x: string, y: string): string;
        |]const a = 1;
      */
-    if (overloadRangeToMove && (statements[overloadRangeToMove.end].end >= range.end || range.end === statements[overloadRangeToMove.end].end + 1)) {
+    if (overloadRangeToMove && (statements[overloadRangeToMove.end].end >= range.end || overloadRangeToMove.end === statements.length - 1 || range.end <= statements[overloadRangeToMove.end + 1].getStart())) {

```

Comment:
Rewriting these conditions with some substitutions, this should be equivalent:

```ts
range.end <= statements[N].end ||
range.end <= statements[N + 1].getStart()
```

where `N` is `overloadRangeToMove.end`. Since `statements` is in source order, it sure looks like the first condition is fully subsumed by the second. A statement’s start position can’t possibly be smaller than the previous statement’s end position.

Reply:
I do think you still need to check for if `N + 1` is still valid though, right?

Also, aside, can we rename `.end` to `lastStatementIndex` or something more descriptive?

Reply:
Yes, I was omitting the length check for simplicity. I might write this something like

```ts
const statementAfterLastOverload = overloadRangeToMove && statements[overloadRangeToMove.end + 1];
if (statementAfterLastOverload && range.end <= statementAfterLastOverload()) {
  // ...
}
```

as I think that shows the intent more clearly than the length check.